projects/extension/src/components/Settings.tsx

```diff
@@ -0,0 +1,277 @@
+    // thus its called only during page construction and is saved in a state variable
+    chrome.runtime.getBackgroundPage((backgroundPage) => {
+      const background = backgroundPage as Background
+      setBg(background)
+      setDefaultWellKnownChainBn(
+        background?.uiInterface.getDefaultBootnodes(selectedChain),
+      )
+    })
+  }, [selectedChain])
+
+  useEffect(() => {
+    const getBootnodes = async () => {
+      setBootnodes(await bg?.uiInterface.wellKnownChainBootnodes)
+    }
+    // Whenever the save Button gets disabled (meaning it was called or upon init)
+    // reload the bootnodes from the background script
+    disabledSaveButton && getBootnodes()
+  }, [bg, disabledSaveButton])
+
+  useEffect(() => {
+    const tmpDef: BootnodesType[] = []
+    const tmpCust: BootnodesType[] = []
+    if (bootnodes) {
+      bootnodes[selectedChain].forEach((b) => {
+        const defaultBootnodes =
+          bg?.uiInterface.getDefaultBootnodes(selectedChain)
+        defaultBootnodes?.length && defaultBootnodes?.includes(b)
+          ? tmpDef.push({ bootnode: b, checked: true })
+          : tmpCust.push({ bootnode: b, checked: true })
+      })
+      setDefaultBn(tmpDef)
+      setCustomBn(tmpCust)
+    }
+  }, [bg, bootnodes, selectedChain])
+
+  const alterBootnodes = (
+    bootnode: string,
+    add: boolean,
+    defaultBootnode?: boolean,
+  ) => {
+    const tmp = defaultBootnode ? [...defaultBn] : [...customBn]
+    const i = tmp.findIndex((b) => b.bootnode === bootnode)
+    if (i !== -1) {
+      tmp[i].checked = add
+    } else {
+      tmp.push({ bootnode, checked: true })
+    }
+    defaultBootnode ? setDefaultBn(tmp) : setCustomBn(tmp)
+    disabledSaveButton && setDisabledSaveButton(false)
+  }
+
+  return (
+    <div className="bg-white border border-neutral-200 p-4 rounded-md">
+      {/* Network selection */}
+      <Title titleType="large">Bootnodes</Title>
+      <Title>Network</Title>
+      <div className="networkSelect">
+        <select
+          onChange={(v) => {
+            setSelectedChain(v.target.value)
+            setDisabledSaveButton(true)
+            setCustomBnInput("")
+          }}
+        >
+          <option value="polkadot">Polkadot</option>
+          <option value="ksmcc3">Kusama</option>
+          <option value="westend2">Westend</option>
+          <option value="rococo_v2_2">Rococo</option>
+        </select>
+        <span className="focus"></span>
+      </div>
+      {/* Bootnodes selection */}
+      <Title>Network Bootnodes</Title>
+      <Title titleType="small">Default</Title>
+      <div className="mb-8">
+        {defaultWellKnownChainBn?.map((bn) => (
+          <div className="leading-4 flex items-center mb-2">
+            <CheckBox
+              bootnode={bn}
+              alterBootnodes={alterBootnodes}
+              defaultBootnode={true}
+              isChecked={defaultBn.map((d) => d.bootnode).includes(bn)}
```

Comment:
IMO the bootnode should be "validated" before being added to the list. Checking if a bootnode address is valid takes in the order of milliseconds.
It doesn't make sense for a bootnode to be in the list and invalid. This would be plain confusing.

Reply:
In order for a bootnode to be added to the list, it needs first to be validated. There are 2 lists, the "default bootnodes" one that contain all bootnodes coming from the chainspecs (already validated by us), and an empty "custom bootnodes" one. In the latter, bootnodes can be added only through the "Add" functionality/button which calls the `validateAndAddBootnode` function. This (as per name) validates the bootnode and add it to the custom list. Once user hits "save" then it is added in the localstorage with the "Custom bootnodes" list.

TLDR; All bottnodes in lists with checkboxes are valid - to add one it first gets validated with `unstabe_p2p_discover`